.eslintrc.js

```diff
@@ -0,0 +1,94 @@
+module.exports = {
+  root: true,
+  env: {
+    browser: true,
+    node: true,
+    es6: true,
+    jasmine: true,
+    jest: true,
+  },
+  extends: [
+    '@redhat-cloud-services/eslint-config-redhat-cloud-services',
+    'eslint:recommended',
+    'plugin:import/errors',
+    'plugin:import/warnings',
+    'plugin:import/typescript',
+    'plugin:react/recommended',
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: './tsconfig.json',
+    comment: true,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    ecmaVersion: 2017,
+    extraFileExtensions: ['.json'],
+    sourceType: 'module',
+    tsconfigRootDir: './',
+  },
+  plugins: ['react', 'react-hooks', '@typescript-eslint'],
+  rules: {
+    camelcase: [
+      'error',
+      { allow: ['UNSAFE_componentWillReceiveProps', 'UNSAFE_componentWillMount'] },
+    ],
+    'comma-dangle': ['error', 'always-multiline'],
+    'consistent-return': 0,
+    'consistent-this': [1, 'that'],
+    'default-case': [2],
+    'dot-notation': [2],
+    'no-multiple-empty-lines': [2, { max: 2, maxEOF: 0 }],
+    eqeqeq: [2, 'allow-null'],
+    'guard-for-in': 2,
+    'import/no-duplicates': ['error'],
+    'max-nested-callbacks': [1, 4],
+    'no-alert': 2,
+    'no-caller': 2,
+    'no-console': 2,
+    'no-else-return': ['error'],
+    'no-global-strict': 0,
+    'no-restricted-imports': ['error',
+      {
+        name: '@patternfly/react-icons',
+        message: 'Don\'t use group imports. Use @patternfly/react-icons/dist/js/icons/(kebab-case-name) instead.'
+      },
+      {
+        name: 'lodash',
+        message: 'Don\'t use group imports. Use lodash/(funcName) instead.'
+      }
+    ],
+    'no-shadow': ['error'],
+    'no-underscore-dangle': 0,
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { varsIgnorePattern: 'React', args: 'after-used' },
+    ],
+    '@typescript-eslint/no-use-before-define': 2,
+    'no-var': 2,
+    'object-shorthand': ['error', 'properties'],
+    'prefer-const': ['error', { destructuring: 'all' }],
+    'prefer-template': 2,
+    radix: 2,
+    'react/display-name': 0,
```

Comment:
Even though I sometimes have to disable this rule on some cases, it's good to debug components with their names.

Reply:
Although I agree, it adds a weight you don't necessarily always want. DisplayName DOES help with tracing through the DevTools so that's good... however when you need to just throw an Anonymous function at something just to achieve a goal for a less used case (ie, an error case), you're forced to make more overhead with this rule turned on (ie, removed from our lint rules). 

`DynamicRoute` makes an inline error component, for instance, that inline component doesn't have a name. When debugging it will look odd, but it's not mysterious. I think if we got into the habit of doing inline functions everywhere this would be more problematic and thus desired, but I do not see it as a requirement today.

If you disagree, let me know, I'm not taking a counter position -- I can turn it on and move the aforementioned component to it's own stand-alone component with props et all.

Reply:
Yeah, good point. Mhh, let's keep it as is, we can always turn it on later down the line.